tests/test_qgis_bot.py

```diff
@@ -17,24 +17,92 @@
 #  along with pytest-qgis.  If not, see <https://www.gnu.org/licenses/>.
 #
 import pytest
-from qgis.core import QgsGeometry
+from qgis.core import QgsFieldConstraints, QgsGeometry
 from qgis.gui import QgsAttributeDialog
 
 
-@pytest.mark.parametrize("bot", ["qgis_bot", "module_qgis_bot"])
-def test_create_feature_with_attribute_dialog(layer_points, bot, request):
-    bot = request.getfixturevalue(bot)
+def test_create_simple_feature_with_attribute_dialog(layer_points, qgis_bot):
     layer = layer_points
     count = layer.featureCount()
 
     layer.startEditing()
-    feat = bot.create_feature_with_attribute_dialog(
+    feat = qgis_bot.create_feature_with_attribute_dialog(
         layer, QgsGeometry.fromWkt("POINT(0,0)")
     )
     assert layer.featureCount() == count + 1
     assert feat["bool_field"] is False  # With normal way of creating, it would be NULL.
 
 
+@pytest.mark.parametrize("ignore_warnings", [False, True])
+def test_create_feature_with_attribute_dialog_with_warnings(
+    layer_points, qgis_bot, ignore_warnings
+):
+    layer = layer_points
+    count = layer.featureCount()
+
+    # Set not-null constraint with SOFT strength
+    fields = layer.fields()
+    field_idx = fields.indexOf("text_field")
+    layer.setFieldConstraint(
+        field_idx,
+        QgsFieldConstraints.Constraint.ConstraintNotNull,
+        QgsFieldConstraints.ConstraintStrengthSoft,
+    )
+
+    layer.startEditing()
+
+    if ignore_warnings:
+        feat = qgis_bot.create_feature_with_attribute_dialog(
+            layer, QgsGeometry.fromWkt("POINT(0,0)"), ignore_warnings=ignore_warnings
+        )
+        assert layer.featureCount() == count + 1
+        assert (
+            feat["bool_field"] is False
```

Comment:
Where is this default value False configured for the field? If not set this should be None/null.

Reply:
This value is set by the dialog since the default widget for boolean fields is `QCheckBox` and its default value is unchecked, which is False.